src/temp_sensor.rs

```diff
@@ -0,0 +1,177 @@
+))]
+use esp_idf_sys::soc_periph_temperature_sensor_clk_src_t_TEMPERATURE_SENSOR_CLK_SRC_RC_FAST;
+#[cfg(any(esp32c2, esp32c3, esp32c5, esp32c6, esp32c61, esp32h2))]
+use esp_idf_sys::soc_periph_temperature_sensor_clk_src_t_TEMPERATURE_SENSOR_CLK_SRC_XTAL;
+
+// -- TempSensorClockSource --
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+/// Rust translation of `temperature_sensor_clk_src_t`
+pub enum TempSensorClockSource {
+    Default,
+    #[cfg(any(
+        esp32c2, esp32c3, esp32c5, esp32c6, esp32c61, esp32h2, esp32s2, esp32s3
+    ))]
+    RcFast,
+    #[cfg(any(esp32c2, esp32c3, esp32c5, esp32c6, esp32c61, esp32h2))]
+    XTAL,
+    #[cfg(esp32p4)]
+    LpPeri,
+}
+
+impl From<&TempSensorClockSource> for temperature_sensor_clk_src_t {
+    fn from(value: &TempSensorClockSource) -> Self {
+        match *value {
+            #[cfg(any(
+                esp32c2, esp32c3, esp32c5, esp32c6, esp32c61, esp32h2, esp32s2, esp32s3
+            ))]
+            TempSensorClockSource::RcFast => {
+                soc_periph_temperature_sensor_clk_src_t_TEMPERATURE_SENSOR_CLK_SRC_RC_FAST
+            }
+            #[cfg(any(esp32c2, esp32c3, esp32c5, esp32c6, esp32c61, esp32h2))]
+            TempSensorClockSource::XTAL => {
+                soc_periph_temperature_sensor_clk_src_t_TEMPERATURE_SENSOR_CLK_SRC_XTAL
+            }
+            #[cfg(esp32p4)]
+            TempSensorClockSource::LpPeri => {
+                soc_periph_temperature_sensor_clk_src_t_TEMPERATURE_SENSOR_CLK_SRC_LP_PERI
+            }
+            TempSensorClockSource::Default => {
+                soc_periph_temperature_sensor_clk_src_t_TEMPERATURE_SENSOR_CLK_SRC_DEFAULT
+            }
+        }
+    }
+}
+
+impl From<temperature_sensor_clk_src_t> for TempSensorClockSource {
+    fn from(value: temperature_sensor_clk_src_t) -> Self {
+        match value {
+            #[cfg(any(
+                esp32c2, esp32c3, esp32c5, esp32c6, esp32c61, esp32h2, esp32s2, esp32s3
+            ))]
+            #[allow(non_upper_case_globals)]
+            soc_periph_temperature_sensor_clk_src_t_TEMPERATURE_SENSOR_CLK_SRC_RC_FAST => {
+                Self::RcFast
+            }
+            #[cfg(any(esp32c2, esp32c3, esp32c5, esp32c6, esp32c61, esp32h2))]
+            #[allow(non_upper_case_globals)]
+            soc_periph_temperature_sensor_clk_src_t_TEMPERATURE_SENSOR_CLK_SRC_XTAL => Self::XTAL,
+            #[cfg(esp32p4)]
+            #[allow(non_upper_case_globals)]
+            soc_periph_temperature_sensor_clk_src_t_TEMPERATURE_SENSOR_CLK_SRC_LP_PERI => {
+                Self::LpPeri
+            }
+            // TODO: Perhaps the default value should be mapped explicitly
+            // and all other (u32) values should cause a failure
+            _ => Self::Default,
+        }
+    }
+}
+
+// -- TempSensorConfig --
+pub type TempSensorConfig = config::Config;
+pub mod config {
+    use super::*;
+    #[derive(Debug, Clone, PartialEq, Eq, Hash)]
```

Comment:
Add `#[non_exhaustive]` so that we have backwards compat with future extensions to this struct.

Reply:
Done in https://github.com/esp-rs/esp-idf-hal/pull/337/commits/e32f03db07e5fe4cd63c093e3349011c4bae1478